### PR TITLE
[MIG] contract_payment_mode: Migration to 10.0

### PR DESCRIPTION
**Pull request by chienandalu**
Minor migration tweaks

cc @Tecnativa

### CODE REVIEW

File: contract_payment_mode/__init__.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Antiun Ingenieria S.L. - Antonio Espinosa
```
Comment (pedrobaeza) [inline contract_payment_mode/__init__.py]: Remove copyright here

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
And add the dependency on oca_repositories.txt to bank-payment

File: contract_payment_mode/oca_dependencies.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+bank-payment
```
Comment (pedrobaeza) [inline contract_payment_mode/oca_dependencies.txt]: oca_dependencies.txt file should be in the root, not here.

**Comment by pedrobaeza** [comment]
@tedsalmon @lasley your mock patch in https://github.com/OCA/contract/blob/10.0/product_contract/tests/test_sale_order.py is provoking that our tests fail: https://travis-ci.org/OCA/contract/jobs/236107899#L805. Please unpatch correctly on your tearDown (it's not automatically undone as this is not a DB transaction)

**Comment by lasley** [comment]
@pedrobaeza - why close ths?

**Comment by pedrobaeza** [comment]
Sorry, my fault

**Comment by chienandalu** [comment]
@pedrobaeza  It's all green now

File: contract_payment_mode/models/contract.py [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        domain=[('payment_type', '=', 'inbound')],
+    )
+
+    @api.onchange('partner_id')
+    def on_change_partner_id(self):
+        if self.partner_id.customer_payment_mode_id:
+            self.payment_mode_id = self.partner_id.customer_payment_mode_id.id
+
+    @api.model
+    def _prepare_invoice_data(self, contract):
```
Comment (pedrobaeza) [inline contract_payment_mode/models/contract.py]: As we have seen on v9, this method is incorrect. Correct one is `_prepare_invoice` (see https://github.com/OCA/contract/pull/73/files#diff-237fb5082150850f78bb85c3c23bf563R20)

File: contract_payment_mode/tests/test_contract_payment.py [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+            'recurring_interval': 1,
+            'recurring_invoice_line_ids': [(0, 0, {
+                'quantity': 2.0,
+                'price_unit': 200.0,
+                'name': 'Database Administration 25',
+                'product_id': self.product.id,
+                'uom_id': self.product.uom_id.id,
+            })]
+        })
+        res = self.contract._prepare_invoice_data(self.contract)
```
Comment (pedrobaeza) [inline contract_payment_mode/tests/test_contract_payment.py]: Test shouldn't call directly this method, but check result on the invoice after calling `recurring_create_invoice`. Check https://github.com/OCA/contract/pull/73/files#diff-fad070bd2b05e275bf84e1c69de78209L72

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by chienandalu** [comment]
@pedrobaeza Changes done and green already

**Review by pedrobaeza — APPROVED** [review]

**Comment by pedrobaeza** [comment]
@rafaelbn please review

**Review by rafaelbn — APPROVED** [review]
Tested in runbot 👍

**Pull request merged**